x-pack/plugins/ingest_manager/server/services/index.ts

```diff
@@ -24,12 +24,38 @@ export interface ESIndexPatternService {
  * A service that provides exported functions that return information about an Agent
  */
 export interface AgentService {
+  /**
+   * Get an Agent by id
+   */
+  getAgent(soClient: SavedObjectsClientContract, agentId: string): Promise<Agent>;
+  /**
+   * Authenticate an agent with access toekn
+   */
+  authenticateAgentWithAccessToken(
+    soClient: SavedObjectsClientContract,
+    request: KibanaRequest
+  ): Promise<Agent>;
```

Comment:
Why do we need this method and why do we pass the whole request? Can we pass what is needed incase the request is not compatible from Security Solution to Ingest Manager.

Reply:
This method is used for the  endpoint artifact API, to authenticate user, the method is already used it was just not typed correctly